### PR TITLE
Feat/sd jwt implementation

### PR DESCRIPTION
**Pull request by cjhowland**
Adds SD-JWT support for sign and verify endpoints, including optional key binding.

By default, all claims at all levels of the payload can be selectively disclosable (with the exception of essential verification data such as iss, iat, cnf, etc.), unless indicated otherwise by the issuer.

Info from @dbluhm: This PR, like the JWT sign and verify endpoints added previously, adds basic support for creation and verification of SD-JWTs. The methods/endpoints do not place any expectations on the payload beyond those required by the SD-JWT specification; in other words, the payload is not required to be a VC. As it stands, this enables other services to take advantage of ACA-Py's secure storage and DID Resolution capabilities to create and verify SD-JWT VCs but does not enable ACA-Py to do the same on its own. This is the foundational work required to later add support for SD-JWT VCs to the rest of ACA-Py's Issuance and Verification stacks.

### CODE REVIEW

**Comment by swcurran** [comment]
What a treat on a Friday afternoon!!  Way cool!

Would be great to see a doc file included that outlined the functionality.

File: pyproject.toml [hunk 1]
```diff
@@ -51,6 +51,7 @@ unflatten="~0.1"
 asyncpg = ">=0.25.0,<0.26.0"
 web-py = ">=0.62,<1.0"
 pygments = ">=2.10,<3.0"
+sd_jwt = {git = "https://github.com/openwallet-foundation-labs/sd-jwt-python.git"}
```
Comment (dbluhm) [inline pyproject.toml:54]: We should nudge the OWF to publish a package to PyPI.

Comment (dbluhm) [reply pyproject.toml:54]: Do we have any contacts over there?

Comment (dbluhm) [reply pyproject.toml:54]: https://github.com/openwallet-foundation-labs/sd-jwt-python/issues/6

File: aries_cloudagent/wallet/sd_jwt.py [hunk 2]
```diff
@@ -0,0 +1,320 @@
+"""Operations supporting SD-JWT creation and verification."""
+
+import re
+from typing import Any, List, Mapping, Optional, Union
+from marshmallow import fields
+from jsonpath_ng.ext import parse
```
Comment (andrewwhitehead) [inline aries_cloudagent/wallet/sd_jwt.py:6]: I had to look up what `parse` was referring to, it might be better to use an alias here (`jsonpath_parse` for instance).

File: aries_cloudagent/wallet/tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import pytest
```
Comment (andrewwhitehead) [inline aries_cloudagent/wallet/tests/conftest.py:1]: This file doesn't seem related to the rest?

**Review by dbluhm — APPROVED** [review]

**Review by andrewwhitehead — APPROVED** [review]

**Comment by andrewwhitehead** [comment]
Looks comprehensive, thanks! Hopefully that library does get released on PyPI shortly.
